plugins/glustershd/rest.go

```diff
@@ -124,3 +130,100 @@ func selfhealInfoHandler(w http.ResponseWriter, r *http.Request) {
 	restutils.SendHTTPResponse(ctx, w, http.StatusOK, &info.Bricks)
 
 }
+
+func selfHealHandler(w http.ResponseWriter, r *http.Request) {
+	// Collect inputs from URL
+	p := mux.Vars(r)
+	volname := p["name"]
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	var healType string
+	heal, ok := r.URL.Query()["type"]
+	if ok {
+		switch heal[0] {
+		case "index":
+			healType = indexHeal
+		case "full":
+			healType = fullHeal
+		default:
+			restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "heal type can only be either index or full")
+			return
+		}
+	} else {
+		healType = indexHeal
+	}
+
+	txn, err := transaction.NewTxnWithLocks(ctx, volname)
+	if err != nil {
+		if err == transaction.ErrLockTimeout {
+			restutils.SendHTTPError(ctx, w, http.StatusConflict, err)
+		} else {
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		}
+		return
+	}
+	defer txn.Done()
+
+	// Validate volume existence
+	volinfo, err := volume.GetVolume(volname)
+	if err != nil {
+		if err == gderrors.ErrVolNotFound {
+			restutils.SendHTTPError(ctx, w, http.StatusNotFound, gderrors.ErrVolNotFound)
+		} else {
+			restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		}
+		return
+	}
+
+	// Check if volume is started
+	if volinfo.State != volume.VolStarted {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, gderrors.ErrVolNotStarted)
+		return
+	}
+
+	// Check if self heal is already enabled
+	if !isHealEnabled(volinfo) {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "self heal option is disabled for this volume")
+		return
+	}
+
+	if err := txn.Ctx.Set("volname", volname); err != nil {
+		logger.WithError(err).WithField(
+			"key", "volname").Error("failed to set key in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
+	if err := txn.Ctx.Set("volinfo", volinfo); err != nil {
+		logger.WithError(err).WithField("key", "volinfo").Error("failed to set key in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
+	if err := txn.Ctx.Set("healType", healType); err != nil {
+		logger.WithError(err).WithField("key", "healType").Error("failed to set key in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+
+	txn.Nodes = volinfo.Nodes()
+	txn.Steps = []*transaction.Step{
+		{
+			DoFunc: "selfheal.Heal",
+			Nodes:  txn.Nodes,
+		},
+	}
+
+	err = txn.Do()
+	if err != nil {
+		logger.WithFields(log.Fields{
+			"error":   err.Error(),
+			"volname": volname,
+		}).Error("failed to start heal process")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
+		return
+	}
+	restutils.SendHTTPResponse(ctx, w, http.StatusOK, "Heal on volume has been successfully launched. Use heal info to check status")
```

Comment:
This should have no message in REST. Just pass `nil`.

You can display such a message in CLI